buildSrc/src/main/groovy/org/gradle/build/docs/ExtractSnippetsTask.groovy

```diff
@@ -15,38 +15,67 @@
  */
 package org.gradle.build.docs
 
+import org.gradle.api.Action
+import org.gradle.api.DefaultTask
 import org.gradle.api.file.FileTree
 import org.gradle.api.file.FileVisitDetails
 import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.OutputDirectory
 import org.gradle.api.tasks.PathSensitive
 import org.gradle.api.tasks.PathSensitivity
-import org.gradle.api.tasks.OutputDirectory
-import org.gradle.api.tasks.SourceTask
 import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.util.PatternFilterable
 
 import java.util.regex.Matcher
 import java.util.regex.Pattern
+
 /**
  * Produces the snippets files for a set of sample source files.
  */
 @CacheableTask
-class ExtractSnippetsTask extends SourceTask {
+class ExtractSnippetsTask extends DefaultTask {
 
     @OutputDirectory
     File destDir
 
     @OutputDirectory
     File snippetsDir
 
-    @Override
+    @Input
```

Comment:
Why are the `excludes` an `@Input`? Wouldn't a change to those directly affect the `source` property? And if it does not affect the source property, the task should stay up-to-date I guess.

Reply:
yep, you're right.